File: spring-boot-async/src/main/java/com/example/async/domain/order/Order.java
package com.example.async.domain.order;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
@Entity
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "product_id")
    private Long productId;

    public Order createOrder(long productId) {
        Order order = new Order();
        order.productId = productId;

        return order;
    }
}

File: spring-boot-mysql/src/main/kotlin/com/example/demo/bar/order/Order.kt
package com.example.demo.bar.order

import com.example.demo.bar.person.Person
import java.time.Instant
import javax.persistence.*

@Entity
@Table(name = "orders")
class Order(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,

    var createdAt: Instant = Instant.now(),

    @ManyToOne(fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
    @JoinColumn(name = "person_id")
    var person: Person
)